soh/soh/OTRGlobals.cpp

```diff
@@ -1745,6 +1773,9 @@ extern "C" int CustomMessage_RetrieveIfExists(GlobalContext* globalCtx) {
             }
         } else if (textId >= 0x9000 && textId <= 0x905F) {
             messageEntry = Randomizer_GetScrubMessage((textId & ((1 << 8) - 1)));
+        } else if (textId >= 0x9100 && textId <= 0x9180) {
+            // TODO: Magic number 100, we don't to overwrite deku scrub messages
+            messageEntry = Randomizer_GetShopMessage((textId & ((1 << 8) - 1)) + 100);
```

Comment:
Not really happy about the flow here from call to call with the bitwise stuff and avoiding conflicts, brainstorming on a better solution

Reply:
short term it's probably worth it to add a detailed comment explaining the logic (and replace the magic number with a useful name)